library/src/main/scala/scala/scalajs/runtime/Semantics.scala

```diff
@@ -0,0 +1,40 @@
+package scala.scalajs.runtime
+
+import scala.annotation.switch
+
+/** Utilities to test for erroneous conditions depending on the Semantics
+ *  configuration.
+ */
+object Semantics {
+
+  // Must be in sync with checkedBehaviorToInt() in OptimizerCore
+  private final val Compliant = 0
+  private final val Fatal = 1
+  private final val Unchecked = 2
+
+  // Note: can return `Unchecked` if caught as an intrinsic by the optimizer
+  private def asInstanceOfs: Int =
+    if (environmentInfo.semantics.compliantAsInstanceOfs) Compliant
+    else Fatal
```

Comment:
Since we will have to add an IR node for the `LinkingInfo` anyways, I'd recommend to make the `LinkingInfo` a compiler primitive to avoid this duplicate code.